### PR TITLE
Platypus with yyPTP

### PR DESCRIPTION
**Pull request by midnight-commit**

### CODE REVIEW

File: contracts/strategies/PlatypusVoterProxy.sol [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        _;
+    }
+
+    modifier onlyStrategy(address _asset) {
+        require(strategies[_asset] == msg.sender, "PlatypusVoterProxy:onlyStrategy");
+        _;
+    }
+
+    // asset => strategies
+    mapping(address => address) public strategies;
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoterProxy.sol:56]: In theory, there could be a second strategy with the same asset. This mapping means we need to be careful if we redeploy any strategies with the same token. Does not need to be fixed now, but we should document

File: contracts/strategies/PlatypusVoterProxy.sol [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                abi.encodeWithSignature("transfer(address,uint256)", stakerFeeReceiver, stakingFee)
+            );
+        }
+
+        uint256 reward = pendingPtp.sub(boostFee).sub(stakingFee);
+
+        platypusVoter.safeExecute(PTP, 0, abi.encodeWithSignature("transfer(address,uint256)", msg.sender, reward));
+
+        if (bonusTokenAddress > address(0)) {
+            platypusVoter.wrapAvaxBalance();
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoterProxy.sol:236]: is this supposed to handle AVAX/WAVAX as a bonus reward? if so, should it be outside of this clause?

File: contracts/strategies/PlatypusVoterProxy.sol [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        platypusVoter.safeExecute(
+            _stakingContract,
+            0,
+            abi.encodeWithSignature("deposit(uint256,uint256)", _pid, liquidity)
+        );
+    }
+
+    function reinvestFeeBips() external returns (uint256) {
+        uint256 boostFee = 0;
+        if (boosterFee > 0 && boosterFeeReceiver > address(0) && platypusVoter.depositsEnabled()) {
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoterProxy.sol:120]: `platypusVoter.depositsEnabled()` nice touch

Comment (snow-farmer) [reply contracts/strategies/PlatypusVoterProxy.sol:120]: the fee could be >0 even if deposits are not enabled. PS - not seeing this fn in platypusVoter

File: contracts/strategies/PlatypusVoter.sol [hunk 4]
```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.3;
+
+import "../interfaces/IWAVAX.sol";
+import "../interfaces/IVePTP.sol";
+import "../lib/SafeERC20.sol";
+import "../lib/Ownable.sol";
+import "../lib/ERC20.sol";
+
+contract PlatypusVoter is Ownable, ERC20 {
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoter.sol:10]: may make sense to use with YakERC20 instead (https://github.com/yieldyak/smart-contracts/blob/master/contracts/YakERC20.sol)

File: contracts/strategies/PlatypusVoter.sol [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    IWAVAX private constant WAVAX = IWAVAX(0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7);
+    address public constant PTP = address(0x22d4002028f537599bE9f666d1c4Fa138522f9c8);
+    IVePTP public constant vePTP = IVePTP(0x5857019c749147EEE22b1Fe63500F237F3c1B692);
+
+    address public immutable devAddr;
+    address public voterProxy;
+    bool public depositsEnabled;
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoter.sol:20]: introduce toggle for this flag if we are using it here. We also have the capability to do this at the strategy level

File: contracts/strategies/PlatypusVoter.sol [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    function depositFromBalance(uint256 _amount) external onlyPlatypusVoterProxy {
+        require(depositsEnabled == true, "PlatypusVoter:deposits disabled");
+        _deposit(_amount);
+    }
+
+    function _deposit(uint256 _amount) internal {
+        IERC20(PTP).safeApprove(address(vePTP), _amount);
+        vePTP.deposit(_amount);
+        _mint(msg.sender, _amount);
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoter.sol:70]: _mint before deposit

File: contracts/strategies/PlatypusVoter.sol [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        IERC20(PTP).safeApprove(address(vePTP), 0);
+    }
+
+    function wrapAvaxBalance() external onlyPlatypusVoterProxy returns (uint256) {
```
Comment (snow-farmer) [inline contracts/strategies/PlatypusVoter.sol:74]: could all the `onlyPlatypusVoterProxy` restricted functions could be done via `execute`? should the logic sit here?

File: contracts/strategies/PlatypusVoterProxy.sol (hunk 2 above)
Comment (midnight-commit) [reply contracts/strategies/PlatypusVoterProxy.sol:238]: This handles all bonus rewards. It assumes that bonusTokenAddress will be set to WAVAX for WAVAX or AVAX rewards. We could add an extra check if bonusTokenAddress == WAVAX before wrapping AVAX balance.

File: contracts/strategies/PlatypusVoter.sol (hunk 4 above)
Comment (midnight-commit) [reply contracts/strategies/PlatypusVoter.sol:10]: I looked at YakERC20, but the _burn function isn't good for this and you can't change the name and symbol right now.

File: contracts/strategies/PlatypusVoter.sol (hunk 5 above)
Comment (midnight-commit) [reply contracts/strategies/PlatypusVoter.sol:20]: Added, forgot the setter!

File: contracts/strategies/PlatypusVoter.sol (hunk 6 above)
Comment (midnight-commit) [reply contracts/strategies/PlatypusVoter.sol:78]: Doesn't matter with a 1:1 relationship or? Changed it anyways.

**Pull request merged**